src/app/stack/card-details/card-details.component.ts

```diff
@@ -658,22 +835,66 @@ export class CardDetailsComponent implements OnInit, OnChanges {
                 component.ecosystem,
                 this.report.manifest_file_path,
                 null,
-                transitive
+                transitive,
+                component.allTransitiveDependencies,
+                null,
+                component.dependencies,
+                component.url,
+                component.public_vulnerabilities,
+                component.private_vulnerabilities,
+                component.recommended_version,
+                component.vulnerable_dependencies && component.vulnerable_dependencies.length > 0 ? component.vulnerable_dependencies : null,
+                publicSecurityDetails,
+                privateSecurityDetails,
+                this.registrationStatus,
+                false,
+                isTransitive,
+                component.publicTransitiveDependencies,
+                component.privateTransitiveDependencies,
+                this.publicTransitiveCount(component.publicTransitiveDependencies),
+                this.privateTransitiveCount(component.privateTransitiveDependencies),
             );
         }
         return null;
     }
 
+    private publicTransitiveCount(deps: any[]): string {
+        let vulnerabilityCount = 0;
+        if (deps != undefined) {
+            deps.forEach(dep => {
+                vulnerabilityCount += dep.componentInformation.public_vulnerabilities.length;
+            });
+
+            return vulnerabilityCount + '';
```

Comment:
Use toString to convert number to string.
```suggestion
            return vulnerabilityCount.toString();
```

Reply:
fixed